npm/ng-packs/packages/oauth/src/lib/strategies/auth-code-flow-strategy.ts

```diff
@@ -68,4 +69,17 @@ export class AuthCodeFlowStrategy extends AuthFlowStrategy {
     const culture = { culture: lang, 'ui-culture': lang };
     return { ...(lang && culture), ...queryParams };
   }
+
+  private setUICulture() {
+    this.oAuthService.events
+      .pipe(
+        filter(event => event.type === 'token_received'),
+        tap(e => {
+          const urlParams = new URLSearchParams(window.location.search);
+          this.configState.uiCultureFromAuthCodeFlow = urlParams.get('ui-culture');
+          window.history.replaceState(null, '', window.location.pathname);
+        }),
+      )
+      .subscribe();
+  }
```

Comment:
This logic resolves problem, yet we can separate, listening token_received event, set ui-culter and clear url logics. After this enhancement I'll review again